### PR TITLE
Topic/prioritization traits #411

### PR DESCRIPTION
**Pull request by lukewyman**
I've added `private[finch]` to the prioritization traits, and also took a stab at organizing the scaladoc comments into groups.

### CODE REVIEW

File: core/src/main/scala/io/finch/internal/Mapper.scala [hunk 1]
```diff
@@ -14,40 +18,66 @@ trait Mapper[A] {
...
-trait HighPriorityMapperConversions extends LowPriorityMapperConversions {
+
+private[finch] trait HighPriorityMapperConversions extends LowPriorityMapperConversions {
+  /**
+   * @group HighPriorityMapper
+   */
   implicit def mapperFromOutputHFunction[A, B, F, OB](f: F)(implicit
-    ftp: FnToProduct.Aux[F, A => OB],
-    ev: OB <:< Output[B]
+                                                            ftp: FnToProduct.Aux[F, A => OB],
```
Comment (vkostyukov) [inline core/src/main/scala/io/finch/internal/Mapper.scala]: Can we move these two back? I'm surprised scalastyle didn't catch this.

**Comment by vkostyukov** [comment]
Thanks @lukewyman! This is a good start!

**Comment by lukewyman** [comment]
@vkostyukov I've updated the PR with the style corrections you suggested. travis is green.

**Comment by vkostyukov** [comment]
Thanks @lukewyman! Sorry to bother again, but do you mind squashing the PR into a single commit?

**Comment by lukewyman** [comment]
@vkostyukov - currently researching how to do that. will have it soon...

**Comment by vkostyukov** [comment]
Just do `git rebase -i HEAD~4` and mark 3 last commits as `s`. And then `git push -f origin lukewyman:topic/prioritization-traits`.

**Comment by lukewyman** [comment]
@vkostyukov when I did the rebase command, the result shows 13 commits. mark only the last 3 with s?

**Comment by vkostyukov** [comment]
Well, the idea is to merge/squash commits into one. When you mark a commit `s` it will be squashed with the previous one. So basically you need to keep only one of 4 commits (whichever you pick) and squash the rest of them.

**Comment by lukewyman** [comment]
@vkostyukov the thing is that I see 13 commits when i do the rebase command. The 1st and the 13th commits are mine, while the 11 commits inbetween them are not mine.

**Comment by vkostyukov** [comment]
Good news about interactive rebase is that you can change the order of commits by just changing the order of lines.

**Comment by lukewyman** [comment]
@vkostyukov ok, but why are you seeing 4 that need to be squashed. 2 are mine, then what are the other 2?

**Comment by mikejcurry** [comment]
Hey @lukewyman , two commits are the ones you made code changes to. The other two commits are from when you synced up with master, they have titles like: 'Merge branch 'master' into ...' - if you are looking at this PR in the github web interface, you will be able to click on the 'commits' tab at the top of this PR, and you will be able to see the 4 commits referenced.

**Comment by vkostyukov** [comment]
Oh these are just merge commits. I'm happy to help with that and merged it via command line (so I can squash) if that works for you.

**Comment by lukewyman** [comment]
thanks, @vkostyukov, that would be a help. I'm still taking a look tho, so I can understand this in git :)

**Comment by vkostyukov** [comment]
I opened a new PR #543 (with one commit) instead of this one.

**Pull request closed without merging**